packages/blade/src/components/Chip/__tests__/Chip.native.test.tsx

```diff
@@ -0,0 +1,142 @@
+import { fireEvent } from '@testing-library/react-native';
+import React from 'react';
+import { Chip } from '../Chip';
+import { ChipGroup } from '../ChipGroup/ChipGroup';
+import renderWithTheme from '~utils/testing/renderWithTheme.native';
+import { InfoIcon } from '~components/Icons';
+
+beforeAll(() => jest.spyOn(console, 'error').mockImplementation());
+afterAll(() => jest.restoreAllMocks());
+
+describe('<Chip />', () => {
+  it('should render chip', () => {
+    const { toJSON, getByText } = renderWithTheme(
+      <ChipGroup>
+        <Chip value="apple">Apple</Chip>
+        <Chip value="mango">Mango</Chip>
+      </ChipGroup>,
+    );
+
+    expect(toJSON()).toMatchSnapshot();
+    expect(getByText('Apple')).toBeDefined();
+    expect(getByText('Mango')).toBeDefined();
+  });
+
+  it('should render chip with icon', () => {
+    const { toJSON, getByText } = renderWithTheme(
+      <ChipGroup>
+        <Chip value="apple" icon={InfoIcon}>
+          Apple
+        </Chip>
+        <Chip value="mango" icon={InfoIcon}>
+          Mango
+        </Chip>
+      </ChipGroup>,
+    );
+    expect(toJSON()).toMatchSnapshot();
+    expect(getByText('Apple')).toBeDefined();
+    expect(getByText('Mango')).toBeDefined();
+  });
+
+  it('should throw error if used outside ChipGroup', () => {
+    try {
```

Comment:
let's use toThrow. 

```
expect(() =>
      renderWithTheme(
        <Alert
          description="Currently you can only accept payments in international currencies using PayPal."
          // @ts-expect-error testing failure case when there is no primary action passed
          actions={{
            secondary: { text: 'Link', onClick: onClickSecondary },
          }}
        />,
      ),
    ).toThrow(`[Blade: Alert]: SecondaryAction is allowed only when PrimaryAction is defined.`);
```

The issue with try catch is that, if the error never happens (eg: say you remove the runtime check) the catch clause will never run thus test will pass.

Reply:
Updated, I have seen similar usage of try-catch in a few other tests, we should use` toThrow()/toThrowError()` for consistency. maybe we can add an eslint rule for this.

Reply:
Yes that would be nice, if there is a third party eslint rule for this then let's add, otherwise rehne do.